server/channels/app/server.go

```diff
@@ -1152,7 +1152,28 @@ func (a *App) OriginChecker() func(*http.Request) bool {
 
 		return utils.OriginChecker(allowed)
 	}
-	return nil
+
+	// Overriding the default origin checker
+	return func(r *http.Request) bool {
+		origin := r.Header["Origin"]
+		if len(origin) == 0 {
+			return true
+		}
+		u, err := url.Parse(origin[0])
+		if err != nil {
+			return false
+		}
+
+		// To maintain the case where siteURL is not set.
+		if *a.Config().ServiceSettings.SiteURL == "" {
+			return strings.EqualFold(u.Host, r.Host)
+		}
+		siteURL, err := url.Parse(*a.Config().ServiceSettings.SiteURL)
```

Comment:
Should we cover misconfigurations like the siteURL being set to `https:\\example.com\` -- which would presumably work for all cases that don't invoke `net/url.Parse`, but with `net/url` results in an opaque URL.

0/5

Reply:
Hmm .. but then how would you actually parse out the scheme and host in that case? Seems a bit of edge case to me to try to handle that.

Reply:
Yeah probably doesn't make sense trying to parse bad urls like that, but we could try to detect them and bail out:

```go
if siteURL.Host == "" || siteURL.Opaque != "" {
  return false
}
```

Reply:
I am thinking of keep this as is for now to keep things consistent with the rest of the codebase. I don't think we do special handling of what happens if a siteURL is opaque. Either it parses or not. Maybe if the need arises, we handle that all over the codebase in a separate PR?

Reply:
done